### PR TITLE
[Communication] - Autorest - Updating to point to readme.md

### PR DESCRIPTION
**Pull request by JoshuaLai**
Git Issue https://github.com/Azure/azure-sdk-for-net/issues/18733

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Review by ShivangiReja — APPROVED** [review]

**Comment by DominikMe** [comment]
@JoshuaLai Did we test generating code with these changes? The READMEs in the spec repo contain multiple swagger files under the same tag, so I worry this might not work.

**Comment by JoshuaLai** [comment]
> @JoshuaLai Did we test generating code with these changes? The READMEs in the spec repo contain multiple swagger files under the same tag, so I worry this might not work.

The only one that will cause an issue at the moment is the PhoneNumber package cause yes you are right it contains a tag with multiple swaggers. I need to create a PR in the azure-rest-api-specs repo. (https://github.com/Azure/azure-rest-api-specs/pull/13363). I believe this is the correct approach. But I have questions cause it seems like the whole readme seems dated. So should we update it?

File: sdk/communication/Azure.Communication.Identity/src/autorest.md [hunk 1]
```diff
@@ -3,10 +3,12 @@
...
+
 > see https://aka.ms/autorest
 
-``` yaml
-input-file:
-    -  https://raw.githubusercontent.com/Azure/azure-rest-api-specs/5b19c6e69cd2bb9dbe4e5c1237b2c5a175d90ca5/specification/communication/data-plane/Microsoft.CommunicationServicesIdentity/stable/2021-03-07/CommunicationIdentity.json
+```yaml
+tag: package-2021-03-07
+require:
+    - https://github.com/Azure/azure-rest-api-specs/blob/e5a57e87f16c7fd9a6eaeb3c6049293d1334f6c6/specification/communication/data-plane/Microsoft.CommunicationServicesIdentity/readme.md
```
[the remark this answers is not in the document]
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Identity/src/autorest.md:12]: Has the swagger changed between the two commits?

[the remark this answers is not in the document]
Comment (DominikMe) [reply sdk/communication/Azure.Communication.Identity/src/autorest.md:12]: Had an offline chat with Josh.
Yes, the swagger has changed. .NET used an old swagger and this PR now fixes a real bug. The service serializes as `innererror` but the SDK property name is `InnerError`. With the old swagger, the SDK would try to deserialize `innerError` with uppercase E and that will fail.

[the remark this answers is not in the document]
Comment (DominikMe) [reply sdk/communication/Azure.Communication.Identity/src/autorest.md:12]: @RezaJooyandeh We need that casing fix in C# Identity for GA. But we can fix it without @JoshuaLai 's larger README changes and just point to the latest swagger directly. What do you think?

**Comment by JoshuaLai** [comment]
Closing because outdated, phone number was change quite a bit. Going to create a new PR based off of GA

**Pull request closed without merging**